ingestion/src/kldata.rs

```diff
@@ -286,3 +298,174 @@ pub async fn filter_and_label_kldata(
 
     Ok(data)
 }
+
+#[cfg(test)]
+mod tests {
+    use chrono::TimeZone;
+    use test_case::test_case;
+
+    use super::*;
+
+    #[test_case(
+        ("nationalnr=99999", "nationalnr") => Ok(99999); 
+        "parsing nationalnr"
+    )]
+    #[test_case(
+        ("type=508", "type") => Ok(508); 
+        "parsing type"
+    )]
+    #[test_case(
+        ("messageid=23", "messageid") => Ok(23); 
+        "parsing messageid"
+    )]
+    #[test_case(
+        ("unexpected", "messageid") => Err(Error::Parse("unexpected field in kldata header format: unexpected".to_string()));
+        "unexpected field"
+    )]
+    #[test_case(
+        ("unexpected=10", "messageid") => Err(Error::Parse("wrong key in field: expected 'messageid', got 'unexpected'".to_string()));
+        "unexpected key"
+    )]
+    fn test_parse_meta_field((field, key): (&str, &'static str)) -> Result<i32, Error> {
+        parse_meta_field(field, key)
+    }
+
+    #[test_case(
+        "Test message that fails." => Err(Error::Parse("kldata indicator missing or out of order".to_string()));
+        "missing kldata indicator"
+    )]
+    #[test_case(
+        "kldata/nationalnr=100/type=504/messageid=25" => Ok((100, 504, 25));
+        "valid header 1"
+    )]
+    #[test_case(
+        "kldata/nationalnr=99993/type=508/messageid=23" => Ok((99993, 508, 23));
+        "valid header 2"
+    )]
+    #[test_case(
+        "kldata/nationalnr=93140/type=501/add" => Err(Error::Parse("unexpected field in kldata header format: add".to_string()));
```

Comment:
What was the weird case that Ingrid talked about yesterday where we have to delete things? do we handle / test this?

Reply:
It cannot be handled in this PR as it requires changes in obsinn. Issue #21